### ISSUE TITLE
Initial lazy loading (caching to IDB) on Safari doesn't work. Starts working after few attempts to refresh the page

### ISSUE THREAD

**Issue by ksinas** [meteor/meteor#11479]
When you load a web app on Safari (Mac OS) for the first time (incognito mode), lazy components doesn't load. After few refresh attempts it starts loading.

It's certainly a bug in Safari, but I believe Meteor should handle this edge case.

The problem occurs in `dynamic-import/cache.js` when `idb.open` is called. The process hangs and doesn't call `onsuccess` neither `onerror`. Rejecting after some period of time could solve the issue.

I was trying to reproduce the same behaviour just by calling `indexedDB.open('test')` in `client/main.jsx`. But funny enough then everything works perfectly, including `MeteorDynamicImportCache`.

Here's a simple project where it can be reproduced running with --production flag.
https://github.com/ksinas/lazy-load-debug

I added dynamic-import package with some logs which indicates the problem

**Comment by StorytellerCZ** [comment]
Thanks for the report @ksinas . Honestly I don't even know how to properly categorize such annoyance. Do you know if Safari has this logged as a bug or describes this behavior somewhere?

**Comment by ksinas** [comment]
I didn't find anything specific. That's why I wanted to find an easy way to reproduce it on safari, so I could file another bug report. But to my surprise, it was working fine if I would try to call `open` immediately. So it's a really weird issue. Maybe would be interesting to understand why it happens in that particular place of Meteor code and not earlier.

Also, it doesn't happen to me if I first load another page and then type in the address, but it definitely happens if a user clicks on a link in slack for example. Then the page doesn't load properly.

I fixed my production by rejecting after 100ms if it got stuck

**Comment by StorytellerCZ** [comment]
Did you test this with other browsers as well?

**Comment by ksinas** [comment]
Yes. Crome and Firefox works fine. Safari version that I tested myself is 14.1.1. It's easy to reproduce. You can try it yourself with my repo or make your own with this snippet

```js
import React, { lazy, Suspense } from 'react';
import { Meteor } from 'meteor/meteor';
import { render } from 'react-dom';

const App = lazy(() => import('/imports/ui/App'));

Meteor.startup(async () => {
  render((
    <Suspense fallback="Loading...">
      <App />
    </Suspense>
  ), document.getElementById('react-target'));
});
```

**Comment by StorytellerCZ** [comment]
This is probably related to this bug: https://bugs.webkit.org/show_bug.cgi?id=226547

**Comment by ksinas** [comment]
It definitely is. But even if Apple fixes the issue, we can't be certain of the new version adaptation and future regressions. I would still suggest that meteor should schedule a reasonable timeout for the indexedDB to be open.

**Comment by StorytellerCZ** [comment]
@ksinas could you please test the latest version of the fix? Me and @renanccastro did a test on it and it worked, but I would like to extend our sample before we release the fix.

**Comment by StorytellerCZ** [comment]
By the way, this bug is fixed in Safari 14.1

**Comment by ksinas** [comment]
The fix works. But it would have also worked if only one line was added `var idb = global.indexedDB;`. Since the implementation is only targeting Safari, maybe it's an overkill if you can fix it just with this "magic" reference :)

**Comment by ksinas** [comment]
> By the way, this bug is fixed in Safari 14.1

I tested on Safari 14.1.1 and the issue still exists. Maybe it will be fixed in the next release
